src/commands/synthetics/mobile.ts

```diff
@@ -1,27 +1,31 @@
 import * as crypto from 'crypto'
-import {promises as fs} from 'fs'
+import fs from 'fs'
 
 import {APIHelper} from './api'
 import {Test, TestPayload, UserConfigOverride} from './interfaces'
 
-export const getMD5HashFromFileBuffer = async (fileBuffer: Buffer): Promise<string> => {
-  const hash = crypto.createHash('md5').update(fileBuffer).digest('base64')
+export const getMD5HashFromFile = async (file: string): Promise<string> => {
+  const hash = crypto.createHash('md5')
+  const input = fs.createReadStream(file)
+  for await (const chunk of input) {
+    hash.update(chunk)
+  }
 
-  return hash
+  return hash.digest('base64')
 }
 
 export const uploadMobileApplications = async (
   api: APIHelper,
   applicationPathToUpload: string,
   mobileApplicationId: string
 ): Promise<string> => {
-  const fileBuffer = await fs.readFile(applicationPathToUpload)
-  const md5 = await getMD5HashFromFileBuffer(fileBuffer)
+  const md5 = await getMD5HashFromFile(applicationPathToUpload)
   const {presigned_url_params: presignedUrl, file_name: fileName} = await api.getMobileApplicationPresignedURL(
     mobileApplicationId,
     md5
   )
 
+  const fileBuffer = await fs.promises.readFile(applicationPathToUpload)
```

Comment:
Let's use the stream from `fs.createReadStream` in `uploadMobileApplication` instead of the `Buffer` to not manipulate the file with two different formats. It should work out of the box according to the [axios doc](https://axios-http.com/docs/req_config#:~:text=only%3A%20Stream%2C%20Buffer-,data,-%3A%20%7B%0A%20%20%20%20firstName).

Reply:
Wdyt we create a follow-up card for this? We can move forward with the Buffer implementation for now!

Reply:
Created new issue here:
https://datadoghq.atlassian.net/browse/SYNTH-9246

Reply:
Perfect! Thank you!